landlab/components/dimensionless_discharge/dimensionless_discharge.py

```diff
@@ -0,0 +1,196 @@
+
+     to determine whether the dimensionless discharge calculated
+     exceeds thresholds for a sediment-laden (Upper Limit) or
+     water-producing only (Lower Limit) debris flow.  C and N are
+     empirically-derived by comparing dimensionless discharge estimates
+     against previous debris flow events.  The values will vary based on
+     local geology and soil type. In southern California values of C=12
+     and N=0.85 (upper limits, Tang et al, 2019) and C=4.29, N=0.78
+     (lower limits, Tang et al, 2019) have been used, while values of
+     C=0.195 and N=1.27 have been in used in the Italian Dolomites
+     (Gregoretti and Fontana, 2008). Default values are C=12 and N=0.85.
+
+     Parameters
+     ----------
+     soil_density : float list
+         Density of soil in watershed (kg/m^3)
+     water_density : float
+         density of water in waterhed (kg/m^3)
+     C : float
+         Numerator of the debris flow threshold equation; Empirically
+         derived constant (See Tang et al. 2019)
+     N : float
+         Exponent for slope in the denominator of the debris flow
+         threshold equation; Empirically derived constant (See Tang et
+         al. 2019)
+
+     Examples
+     --------
+     >>> from landlab.components import DimensionlessDischarge
+     >>> from landlab import RasterModelGrid
+     >>> import random
+     >>> watershed_grid = RasterModelGrid((3, 3))
+     >>> flux = watershed_grid.add_ones('node', 'flux')
+     >>> d50 = watershed_grid.add_ones('node', 'd50')
+     >>> watershed_grid.at_node ['dem_values'] = \
+     ... np.array([[1.1, 2, 3, 4, 2, 3, 4, 5, 3]])
+     >>> dd = DimensionlessDischarge(watershed_grid)
+     >>> dd.run_one_step()
+     >>> print(watershed_grid.at_node['dimensionless_discharge'])
+     [ 0.55372743  0.55372743  0.55372743  0.55372743  0.55372743
+             0.55372743  0.55372743  0.55372743  0.55372743]
+
+
+
+     References
+     ----------
+     Tang, H., McGuire, L. A., Rengers, F. K., Kean, J. W., Staley,
+     D. M., & Smith, J. B. (2019). Developing and Testing Physically
+     Based Triggering Thresholds for Runoff-Generated Debris Flows.
+     Geophysical Research Letters, 46(15), 8830–8839.
+     https://doi.org/10.1029/2019GL083623
+
+    """
+
+    _name = "DimensionlessDischargeModel"
+
+    _unit_agnostic = False
+
+    _info = {
+        "dimensionless_discharge": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "dimensionless_discharge_above_threshold": {
+            "dtype": bool,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "flux": {
```

Comment:
We'll need a more descriptive name here. Is this field used in another existing component? If so, it should have the same name as that field. Otherwise, it's name should more descriptive name.

Reply:
Agreed. And it may make sense to have this be an existing field name, if possible. As defined in Tang et al, I think q should be thought of as discharge per unit width, as they define it as q=h*(u^2+v^2)^0.5 where h=water depth, and u and v are velocity vectors. 

It would be equivalent to "surface_water__discharge" divided by the relevant face width, I think. That might be a good default, as using it would permit connection to existing components. 

Either that or define a new field that reflects surface water discharge per unit width.

This page may be useful. https://landlab.readthedocs.io/en/master/user_guide/field_io.html

Reply:
@eculler Do you have an opinion on this (will forward to Ryan too)

Reply:
I have changed the name flux to surface_water__discharge

Reply:
@kbarnhart Just had a conversation about this variable. Something like Surface_Water__Width_Normalized_Dischage or Surface_Water__Unit_Dischange would capture what this is a bit better. Is there a variable similar to these already (I couldn't find one)?

Reply:
@mcflugen can you provide guidance here?

Reply:
I'm OK with surface_water__unit_discharge with a description that states that it is volumetric surface water discharge divided by channel width or grid cell width. Also tapping in @gregtucker as he may have thoughts on what it should be called. 

I think that once we have a good name for this field, its metadata is updated, (and the two field-related .rst files are updated to reflect it) that this component is looking good. For example, the metadata has the units as L/T, and I think they should be L^2/T